core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1005,8 +1005,9 @@ private[spark] class SparkSubmit extends Logging {
       case t: Throwable =>
         throw findCause(t)
     } finally {
-      if (args.master.startsWith("k8s") && !isShell(args.primaryResource) &&
-          !isSqlShell(args.mainClass) && !isThriftServer(args.mainClass)) {
+      if (sparkConf.get(AUTO_STOP_ACTIVE_SPARK_CONTEXTS) && args.master.startsWith("k8s") &&
```

Comment:
Shall we put this at the end of this condition?

Reply:
If we put this before `args.master.startsWith("k8s")`, it looks like a general configuration which can be used outside `k8s`.

Reply:
This is a boolean which is easier to evaluate than `String#startsWith` and if it is false the evaluation stops here.

Reply:
This is not a CPU-intensive code part and it doesn't save many CPU cycle. Logical readability is better in this case, @attilapiros .

Reply:
That's true, I move this at the end of the condition.